classes/patreon_frontend.php

```diff
@@ -1073,5 +1074,41 @@ public static function LoginButtonShortcode( $args ) {
 		}
 		
 	}
+	public static function show_patreon_avatar( $avatar, $id_or_email, $size, $default, $alt ) {
+		
+		// Checks if the user has a Patreon avatar saved, and returns that avatar in place of WP/site default
+		$user = false;
+
+		if ( is_numeric( $id_or_email ) ) {
+
+			$id = (int) $id_or_email;
+			$user = get_user_by( 'id' , $id );
+
+		} elseif ( is_object( $id_or_email ) ) {
+
+			if ( ! empty( $id_or_email->user_id ) ) {
+				$id = (int) $id_or_email->user_id;
+				$user = get_user_by( 'id' , $id );
+			}
+
+		} else {
+			$user = get_user_by( 'email', $id_or_email );	
+		}
+
+		if ( $user && is_object( $user ) ) {
+			
+			// Get user's Patreon avatar meta:
+			
+			$user_patreon_avatar = get_user_meta( $user->ID, 'patreon-avatar-url', true );
+			
+			// Override avatar if there is a saved Patreon avatar
+			if ( $user_patreon_avatar != '' ) {
+				$avatar = '<img alt="'.$alt.'" src="'.$user_patreon_avatar.'" class="avatar avatar-"'.$size.' photo" height="'.$size.'" width="'.$size.'" />';
```

Comment:
Can we  is_numeric($size)  and set a default?

Reply:
Since we cannot use size related functions before the remote image is replicated in the local filesystem, is_numeric($size) wouldnt work.

If by setting a default, you mean to distribute a default avatar image to be used for users who log in via Patreon in case they dont have an avatar image at Patreon, yeah, we can do that.

But if you mean setting a default avatar in WordPress if there is no remote Patreon avatar provided, this is not necessary and could interfere in whatever setup site owners have in their WP installations - since WP already has defaults, and there are plugins that bring avatar functionality.

Reply:
I only meant a default size if none is passed in :)
Thanks!

Reply:
Despite in a healthy WP installation this value would always be provided unless the site owner explicitly modified it to false or 0 through manual or plugin modifications, this is a good idea to cover WP installations which may have issues.

Checking for 0 or non numeric and then setting it to wordpress default 48 may cover such problematic installations and make our avatars show. Good thinking.